lib/checkbool.cpp

```diff
@@ -361,7 +364,7 @@ void CheckBool::checkComparisonOfBoolExpressionWithInt()
                      : Token::Match(tok, ">|==|!=")))
                     continue;
                 comparisonOfBoolExpressionWithIntError(tok, true);
-            } else if (isNonBoolStdType(numTok->variable()) && mTokenizer->isCPP())
+            } else if (astIsIntegral(numTok, false) && mTokenizer->isCPP())
```

Comment:
I think I would prefer that `unknown` parameter is `true`. Does that mess up some test case?

Reply:
It seems to trigger a lot of FP:

Testing Complete
Number of tests: 29
Number of todos: 1
Tests failed: 4

../test/testbool.cpp:494(TestBool::comparisonOfBoolExpressionWithInt2): Assertion failed.
Expected:

Actual:
[test.cpp:2]: (warning) Comparison of a boolean expression with an integer.\n

_____
../test/testbool.cpp:936(TestBool::comparisonOfBoolWithInt2): Assertion failed.
Expected:

Actual:
[test.cpp:2]: (warning) Comparison of a boolean expression with an integer.\n

_____
../test/testbool.cpp:982(TestBool::comparisonOfBoolWithInt5): Assertion failed.
Expected:

Actual:
[test.cpp:3]: (warning) Comparison of a boolean expression with an integer.\n

_____
../test/testbool.cpp:709(TestBool::checkComparisonOfFuncReturningBool6): Assertion failed.
Expected:
[test.cpp:6]: (style) Comparison of a function returning boolean value using relational (<, >, <= or >=) operator.\n

Actual:
[test.cpp:6]: (warning) Comparison of a boolean expression with an integer.\n
[test.cpp:6]: (style) Comparison of a function returning boolean value using relational (<, >, <= or >=) operator.\n

Reply:
ah my bad.. of course. I was thinking wrongly.